src/cli/index.js

```diff
@@ -55,8 +55,18 @@ async function main(context) {
     }
   }
 
-  if (context.argv.check && context.argv.listDifferent) {
-    throw new Error("Cannot use --check and --list-different together.");
+  if (
+    [context.argv.check || context.argv.diff, context.argv.listDifferent].filter(
+      Boolean,
+    ).length > 1
+  ) {
+    throw new Error(
+      "Cannot use --check, --list-different, or --diff together."
```

Comment:
Seems like this would be sufficient:

```
if ((context.argv.check || context.argv.diff) && context.argv.listDifferent) {
  throw new error("Cannot use --list-different with --check or --diff");
}
```